### PR TITLE
Definition passing runtime

### PR DESCRIPTION
**Pull request by supunvindula**

### CODE REVIEW

File: test/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
         "build": "ncc build test.js --minify -o dist"
     },
     "dependencies": {
-        "evernode-js-client": "0.6.0"
+        "evernode-js-client": "0.6.0",
+        "xrpl-accountlib": "^2.2.0"
```
Comment (kithminisg) [inline test/package.json:8]: remove this `"xrpl-accountlib": "^2.2.0"` as we are adding that to library.

File: src/xrpl-helpers.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    sign(tx, secret,isMultiSign = false) {
+        const account = derive.familySeed(secret);
+        const signed = sign(tx, (isMultiSign ? [account.signAs(account.address)] : account));
+        return {
+            hash: signed.id,
+            tx_blob: signed.signedTransaction
+        }
+    }
+    multiSign(transactions,isMultiSign=true,accountsArray){
```
Comment (kithminisg) [inline src/xrpl-helpers.js:22]: Remove this multi-sig function, as it is handled in the sign function via an argument.

File: src/xrpl-api.js [hunk 3]
```diff
@@ -412,11 +413,8 @@ class XrplApi {
      * @param {(string | Transaction)[]} transactions An array of signed Transactions (in object or blob form) to combine into a single signed Transaction.
      * @returns A single signed Transaction string which has all Signers from transactions within it.
      */
-    multiSign(transactions) {
-        if (transactions.length > 0) {
-            return xrpl.multisign(transactions);
-        } else
-            throw ("Transaction list is empty for multi-signing.");
+    multiSign(transactions,accountsArray) {
```
Comment (kithminisg) [inline src/xrpl-api.js:416]: Keep this function as it is, as we do not use this for the current ops.

File: src/xrpl-api.js [hunk 4]
```diff
@@ -183,6 +185,8 @@ class XrplApi {
 
         this.#maintainConnection = true;
         await this.#connectXrplClient();
+        const newDefinition = await this.#client.request({ command: 'server_definitions' })
```
Comment (kithminisg) [inline src/xrpl-api.js:188]: newDefinitions

**Review by kithminisg — APPROVED** [review]

File: src/xrpl-api.js [hunk 5]
```diff
@@ -396,11 +396,12 @@ class XrplApi {
...
 
     /**
-     * Submit a single-signature transaction.
+     * Submit a single-signature transaction using tx_blob.
      * @param {string} tx_blob 
      * @returns response object of the submitted transaction.
      */
-    async submit(tx, tx_blob) {
+    async submit(tx_blob) {
+        const tx = this.xrplHelper.decode(tx_blob)
```
Comment (chalith) [inline src/xrpl-api.js:404]: Do we need this decode?

File: src/xrpl-api.js [hunk 6]
```diff
@@ -183,6 +185,8 @@ class XrplApi {
 
         this.#maintainConnection = true;
         await this.#connectXrplClient();
+        const newDefinitions = await this.#client.request({ command: 'server_definitions' })
```
Comment (chalith) [inline src/xrpl-api.js:188]: newDefinitions -> definitions

File: src/xrpl-api.js
Comment (chalith) [inline src/xrpl-api.js:1]: Lets remove `multiSign(transactions)` function since it's not used

**Review by chalith — APPROVED** [review]

**Review by kasunkavinda — APPROVED** [review]

**Pull request merged**